rpc/lighthouse.go

```diff
@@ -42,13 +44,29 @@ func NewLighthouseClient(endpoint string, chainID *big.Int) (*LighthouseClient,
 		endpoint:            endpoint,
 		assignmentsCacheMux: &sync.Mutex{},
 		signer:              signer,
+		lastBlockSeenMux:    &sync.Mutex{},
 	}
 	client.assignmentsCache, _ = lru.New(10)
 
 	return client, nil
 }
 
 func (lc *LighthouseClient) GetNewBlockChan() chan *types.Block {
+	// setup health check & exit if the new block chan does no longer receive new blocks
+	lc.lastBlockSeenMux.Lock()
+	lc.lastBlockSeen = time.Now()
+	lc.lastBlockSeenMux.Unlock()
+	go func() {
+		for ; ; time.Sleep(time.Second) {
+			lc.lastBlockSeenMux.Lock()
+			if time.Since(lc.lastBlockSeen) > time.Minute*2 {
+				// fatal if no new block was received for more than two minutes
+				logger.Fatalf("lighthouse client error, no new block retrieved since %v (%v ago)", lc.lastBlockSeen, time.Since(lc.lastBlockSeen))
```

Comment:
You should unlock the mutex before the fatal.
`lc.lastBlockSeenMux.Unlock()`
As it's not guaranteed that the program will end by the fatal, it can be caught (in the future) and the we end in an deadlock.